### ISSUE TITLE
When the number of classes is very large, the loss function does not converge

### ISSUE THREAD

**Issue by zzzmm1** [kamwoh/orthohash#5]
In face recognition, when the number of classes exceeds 10 million and K=128, the CosFace/ArcFace loss function hardly converges, due to the fixed binary orthogonal codes.
Any suggestions to help with training?
Thanks a lot!

**Comment by kamwoh** [comment]
I think you can try adding some non-linear layers before the hash layer. sometimes it overfits faster so I think it might help to converge.
